File: java-test/src/main/java/post/parthmistry/nst/util/ByteUtil.java
package post.parthmistry.nst.util;

import java.text.DecimalFormat;

public class ByteUtil {

    private static final DecimalFormat decimalFormat = new DecimalFormat();

    static {
        decimalFormat.setMaximumFractionDigits(3);
        decimalFormat.setGroupingUsed(false);
    }

    public static String transformBytes(long bytes) {
        return transformBytes(bytes, (double) bytes);
    }

    public static String transformBytes(double bytes) {

        return bytes + " (" + formatValue((bytes / (1024 * 1024))) + " mb)";
    }

    public static String transformBytes(long bytes, double doubleBytes) {
        return bytes + " (" + formatValue((doubleBytes / (1024 * 1024))) + " mb)";
    }

    public static double bytesToMegaBits(long bytes) {
        return (((double) bytes) / 1_000_000) * 8;
    }

    public static String formatValue(double value) {
        return decimalFormat.format(value);
    }

}

File: rust-test/src/byte_util.rs
pub struct ByteUtil;

impl ByteUtil {

    pub fn transform_bytes(bytes: u64) -> String {
        ByteUtil::transform_bytes_with_f64(bytes, bytes as f64)
    }

    pub fn transform_bytes_with_f64(bytes: u64, double_bytes: f64) -> String {
        format!("{} ({} mb)", bytes, ByteUtil::format_value(double_bytes / (1024.0 * 1024.0)))
    }

    pub fn transform_bytes_f64(bytes: f64) -> String {
        format!("{} ({} mb)", bytes, ByteUtil::format_value(bytes / (1024.0 * 1024.0)))
    }

    pub fn bytes_to_megabits(bytes: u64) -> f64 {
        (bytes as f64 / 1_000_000.0) * 8.0
    }

    pub fn format_value(value: f64) -> String {
        format!("{:.3}", value)
    }

}